File: tests/human/x/fs/order_by_map.fs
let data = [ {| a = 1; b = 2 |}; {| a = 1; b = 1 |}; {| a = 0; b = 5 |} ]
let sorted = data |> List.sortBy (fun x -> (x.a, x.b))
printfn "%A" sorted

File: tests/machine/x/pl/order_by_map.pl
:- style_check(-singleton).
:- initialization(main, main).
main :-
    dict_create(_V0, map, [a-1, b-2]),
    dict_create(_V1, map, [a-1, b-1]),
    dict_create(_V2, map, [a-0, b-5]),
    Data = [_V0, _V1, _V2],
    findall(_V3, (member(X, Data), true, _V3 = X), _V4),
    Sorted = _V4,
    write(Sorted),
    nl,
    true.
